Rakefile

```diff
@@ -1,4 +1,24 @@
-require "bundler/gem_tasks"
-task :default => :spec
+require 'bundler/gem_tasks'
+require 'rspec/core/rake_task'
+
+Dir['tasks/**/*.rake'].each { |rake| load rake }
+
+RSpec::Core::RakeTask.new(:spec)
 
-Dir.glob('tasks/**/*.rake').each(&method(:import))
+task :spec => :compile
+case RUBY_PLATFORM
+when 'java'
+  require 'rake/javaextensiontask'
+  Rake::JavaExtensionTask.new 'murmurhash' do |ext|
+    ext.lib_dir = 'lib/murmurhash'
+    ext.target_version = '1.7'
```

Comment:
is this = to 1.7 or can it be greater than 1.7?

what's the minimum version of java we can support?

Reply:
I had compilation issues with default version (1.6)

Reply:
is 1.7 the minimum version? or does it have to match the exact version?
I want to make sure it will work with 1.8 as well.

Reply:
It is the minimum version. By default it was compiling for 1.5, but there were compilation issues and error for 1.5 and warnings for 1.6:

1.5:

```
javac  -target 1.5 -source 1.5 -Xlint:unchecked  -cp "/Users/max/.rbenv/versions/jruby-9.1.15.0/lib/jruby.jar:" -d tmp/java/murmurhash ext/murmurhash/MurmurHash3.java
warning: [options] bootstrap class path not set in conjunction with -source 1.5
error: Source option 1.5 is no longer supported. Use 1.6 or later.
error: Target option 1.5 is no longer supported. Use 1.6 or later.
```

1.6:

```
javac  -target 1.6 -source 1.6 -Xlint:unchecked  -cp "/Users/max/.rbenv/versions/jruby-9.1.15.0/lib/jruby.jar:" -d tmp/java/murmurhash ext/murmurhash/MurmurHash3.java
warning: [options] bootstrap class path not set in conjunction with -source 1.6
warning: [options] source value 1.6 is obsolete and will be removed in a future release
warning: [options] target value 1.6 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
```

So, I decided to use minimum of 1.7.